### PR TITLE
add new keyboard scatter42 by bbrfkr

### PR DESCRIPTION
**Pull request by bbrfkr**
Addition of new keyboard scatter42 developed by bbrfkr.

## Description

I have debeloped new keyboard "scatter42". This MR contains the default keymap of this keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by bbrfkr** [comment]
Can anyone please review it? Thanks in advance!

**Review by drashna — APPROVED** [review]

File: keyboards/scatter42/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    _RAISE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
```
Comment (fauxpark) [inline keyboards/scatter42/keymaps/default/keymap.c:31]:
```suggestion
```

File: keyboards/scatter42/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+    }
+    return true;
+}
```
Comment (fauxpark) [inline keyboards/scatter42/keymaps/default/keymap.c:59]:
```suggestion
```

File: keyboards/scatter42/scatter42.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  ) \
+  { \
+    { L00, L01, L02, L03, L04 }, \
+    { L10, L11, L12, L13, L14 }, \
+    { L20, L21, L22, L23, L24 }, \
+    { L30, L31, L32, L33, L34, L35 }, \
+    { R04, R03, R02, R01, R00 }, \
+    { R14, R13, R12, R11, R10 }, \
+    { R24, R23, R22, R21, R20 }, \
+    { R35, R34, R33, R32, R31, R30 } \
```
Comment (fauxpark) [inline keyboards/scatter42/scatter42.h:43]:
```suggestion
    { L00, L01, L02, L03, L04, KC_NO }, \
    { L10, L11, L12, L13, L14, KC_NO }, \
    { L20, L21, L22, L23, L24, KC_NO }, \
    { L30, L31, L32, L33, L34, L35 }, \
    { R04, R03, R02, R01, R00, KC_NO }, \
    { R14, R13, R12, R11, R10, KC_NO }, \
    { R24, R23, R22, R21, R20, KC_NO }, \
    { R35, R34, R33, R32, R31, R30 } \
```

File: keyboards/scatter42/config.h [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
+#define PRODUCT_ID   0x0000
```
Comment (fauxpark) [inline keyboards/scatter42/config.h:24]:
```suggestion
#define PRODUCT_ID   0x3B47
```
```
echo -n "scatter42" | sha1sum | cut -c1-4
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by bbrfkr** [comment]
@fauxpark Thank you for your review! I accepted your requests!!

**Review by fauxpark — APPROVED** [review]

**Comment by bbrfkr** [comment]
Thanks for reviewing and merging!!

**Pull request merged**